File: T2/EB_BDD_GR9/02.SOAP_DOTNET_EUREKABANK_GR09/02.SOAP_DOTNET_EUREKABANK_GR09/02.CLIESC/BDD_SOAP_DOTNET_CLIESC_G09/BDD_SOAP_DOTNET_CLIESC_G09/ec/edu/monster/controlador/Desktop_Controlador.cs
using System.Collections.Generic;
using BDD_SOAP_DOTNET_CLIESC_G09.ec.edu.monster.modelo;
using BDD_SOAP_DOTNET_CLIESC_G09.ec.edu.monster.servicio;

namespace BDD_SOAP_DOTNET_CLIESC_G09.ec.edu.monster.controlador
{
    /// <summary>
    /// Controlador que actúa como intermediario entre la vista y el servicio
    /// </summary>
    public class Desktop_Controlador
    {
        private readonly EurekaService service;

        public Desktop_Controlador()
        {
            this.service = new EurekaService();
        }

        /// <summary>
        /// Obtiene los movimientos de una cuenta
        /// </summary>
        public List<Movimiento> TraerMovimientos(string cuenta)
        {
            return service.TraerMovimientos(cuenta);
        }

        /// <summary>
        /// Registra un depósito
        /// </summary>
        public int RegDeposito(string cuenta, double importe)
        {
            return service.RegDeposito(cuenta, importe);
        }

        /// <summary>
        /// Registra un retiro
        /// </summary>
        public int RegRetiro(string cuenta, double importe)
        {
            return service.RegRetiro(cuenta, importe);
        }

        /// <summary>
        /// Registra una transferencia
        /// </summary>
        public int RegTransferencia(string cuentaOrigen, string cuentaDestino, double importe)
        {
            return service.RegTransferencia(cuentaOrigen, cuentaDestino, importe);
        }

        /// <summary>
        /// Valida el ingreso de un usuario
        /// </summary>
        public string ValidarIngreso(string usuario, string password)
        {
            return service.ValidarIngreso(usuario, password);
        }
    }
}

File: T2/EB_BDD_GR9/01.SOAP_JAVA_EUREKABANK_GR09/01.SOAP_JAVA_EUREKABANK_GR09/02.CLIESC/BDD_SOAP_JAVA_CLIESC_G09/src/main/java/ec/edu/monster/controlador/Desktop_Controlador.java
/*
 * Controlador para el cliente de escritorio EurekaBank
 * @author Grupo Monster G09
 */
package ec.edu.monster.controlador;

import ec.edu.monster.modelo.Movimiento;
import ec.edu.monster.servicio.EurekaService;
import java.util.List;

/**
 * Controlador que actúa como intermediario entre la vista y el servicio
 */
public class Desktop_Controlador {
    
    private final EurekaService service;
    
    public Desktop_Controlador() {
        this.service = new EurekaService();
    }
    
    /**
     * Obtiene los movimientos de una cuenta
     * @param cuenta Número de cuenta
     * @return Lista de movimientos
     */
    public List<Movimiento> traerMovimientos(String cuenta) {
        return service.traerMovimientos(cuenta);
    }
    
    /**
     * Registra un depósito
     * @param cuenta Número de cuenta
     * @param importe Importe a depositar
     * @return 1 si fue exitoso, 0 en caso contrario
     */
    public int regDeposito(String cuenta, double importe) {
        return service.regDeposito(cuenta, importe);
    }
    
    /**
     * Registra un retiro
     * @param cuenta Número de cuenta
     * @param importe Importe a retirar
     * @return 1 si fue exitoso, 0 en caso contrario
     */
    public int regRetiro(String cuenta, double importe) {
        return service.regRetiro(cuenta, importe);
    }
    
    /**
     * Registra una transferencia
     * @param cuentaOrigen Cuenta origen
     * @param cuentaDestino Cuenta destino
     * @param importe Importe a transferir
     * @return 1 si fue exitoso, 0 en caso contrario
     */
    public int regTransferencia(String cuentaOrigen, String cuentaDestino, double importe) {
        return service.regTransferencia(cuentaOrigen, cuentaDestino, importe);
    }
}

